### PR TITLE
Feat/filters wearables

### PR DESCRIPTION
**Pull request by agusaldasoro**
Add fields for filtering and pagination to the /wearables endpoint.

### CODE REVIEW

File: lambdas/src/apis/collections/controllers/wearables.ts [hunk 1]
```diff
@@ -120,7 +123,18 @@ export async function getWearables(
...
 
-  return paginatedOffChainWearables.concat(onChainWearables)
+  let moreData: boolean = false
+  if (offChainWearables.length > (onChainPagination.offset + 1) * onChainPagination.limit) {
+    moreData = true
+  }
+
+  const allPagination = { ...onChainPagination, moreData: moreData }
+
+  return {
```
Comment (menduz) [inline lambdas/src/apis/collections/controllers/wearables.ts:133]: we can't break backwards compatibility with people using this endpoint. I understand it changed the schema of the response. 

We should have another new endpoint to handle this filtered case.

Comment (nchamo) [reply lambdas/src/apis/collections/controllers/wearables.ts:133]: This endpoint has not been released, nor has it been deployed to a dev catalyst. We can still make breaking changes

Comment (menduz) [reply lambdas/src/apis/collections/controllers/wearables.ts:133]: Thanks Nico

File: lambdas/src/apis/collections/controllers/wearables.ts [hunk 2]
```diff
@@ -79,48 +79,87 @@ export async function getWearablesEndpoint(
...
+  )
+
+  // Check if there is moreData and then slice for the correct limit
+  const allWearablesWithMore = paginatedOffChainWearablesWithMore.concat(onChainWearablesWithMore)
+  const moreData: boolean = allWearablesWithMore.length > pagination.limit
+  const allWearables = allWearablesWithMore.slice(0, pagination.limit)
+
+  return {
+    wearables: allWearables,
+    filters: filters,
```
Comment (nchamo) [inline lambdas/src/apis/collections/controllers/wearables.ts:141]:
```suggestion
    filters,
```

File: lambdas/src/apis/collections/controllers/wearables.ts [hunk 3]
```diff
@@ -79,48 +79,87 @@ export async function getWearablesEndpoint(
...
 export async function getWearables(
   filters: WearablesFilters,
   pagination: WearablesPagination,
   client: SmartContentClient,
   theGraphClient: TheGraphClient,
   offChainManager: OffChainWearablesManager
-): Promise<Wearable[]> {
+): Promise<{
+  wearables: Wearable[]
+  filters: { collectionIds?: string[]; wearableIds?: string[]; textSearch?: string }
```
Comment (nchamo) [inline lambdas/src/apis/collections/controllers/wearables.ts:110]: Why not use `WearablesFilters`?

File: lambdas/src/apis/collections/controllers/wearables.ts [hunk 4]
```diff
@@ -79,48 +79,87 @@ export async function getWearablesEndpoint(
...
-    res.send(result)
+    const response = await getWearables(requestFilters, sanitizedPagination, client, theGraphClient, offChainManager)
+    res.send(response)
   } catch (error) {
     res.status(500).send(error.message)
   }
 }
 
+/**
+ * As Base Wearables are off chain wearables, we need to render both off-chain and on-chain wearables in the same collection.
```
Comment (nchamo) [inline lambdas/src/apis/collections/controllers/wearables.ts:98]:
```suggestion
 * As Base Wearables are off chain, we need to return both off-chain and on-chain wearables in the same query.
```

**Review by nchamo — APPROVED** [review]
Great work!

**Pull request merged**